adaptors/security_policy_manager_adaptor.py

```diff
@@ -0,0 +1,93 @@
+    ``undeploy()`` and ``cleanup()``. Accepts as parameters an **adaptor_id**
+    (required) and a **template** (optional). The ``translate()`` and ``update()``
+    methods require both an **adaptor_id** and **template**. The ``execute()``,
+    ``undeploy()`` and ``cleanup()`` methods require only the **adaptor_id** .
+
+    :param string adaptor_id: The generated ID of the current application stack
+    :param template: The ADT / ToscaTemplate of the current application stack
+    :type template: ToscaTemplate <toscaparser.tosca_template.ToscaTemplate>
+
+    Usage:
+        >>> from security_policy_manager_adaptor import SecurityPolicyManagerAdaptor
+        >>> se_adapt = SecurityPolicyManagerAdaptor(<adaptor_id>, <ToscaTemplate>)
+        >>> se_adapt.translate()
+            (does nothing but need to implement the abstract function anyways)
+        >>> se_adapt.execute()
+            (send the informations to the security policy manager)
+        >>> se_adapt.update()
+            (update the security policy manager with the new secret)
+        >>> se_adapt.undeploy()
+            (send the security policy manager a undeploy request)
+        >>> se_adapt.cleanup()
+            (does nothing as no files were created but need to implement the abstract function anyways)
+    """
+
+    def __init__(self, adaptor_id, config, template=None):
+        """ Constructor method of the Adaptor as described above """
+        super().__init__()
+        if template and not isinstance(template, ToscaTemplate):
+            raise AdaptorCritical("Template is not a valid TOSCAParser object")
+        self.tpl = template
+        self.ID = adaptor_id
+        self.config = config
+        self.endpoint = 'http://10.97.170.199:5003/'
+        'v1.0/nodecerts'
+        self.status = "init"
+        if template is not None:
+            self.policies = self.tpl.policies
+        logger.debug("Initialising the SE adaptor with the ID and TPL")
+
+    def translate(self):
+        pass
+
+    def execute(self):
+        self.status = "executing"
+        """ Send to the Security Enforcer the informations
+            retrieved from the TOSCA template"""
+        for policy in self.policies:
+            if "tosca.policies.KubernetesSecretDistribution" in policy.type:
+                _interm_dict = policy.get_properties()["text_secrets"].value
+                for key, value in _interm_dict.items():
+                    if self.config["dry_run"] is True:
+                        logger.info("launch api command with params name={} and value={} to {}/v1.0/appsecrets".format(key,value,self.endpoint))
+                    elif self.config["dry_run"] is False:
+                        data_keys = {'name':key, 'value':value}
+                        logger.info("launch secret")
+                        response = requests.post("{}/v1.0/appsecrets".format(self.endpoint), data = data_keys)
+        self.status = "executed"
+
+    def undeploy(self):
```

Comment:
We should implement a function for deleting all secrets in the case the application is undeployed.